### ISSUE TITLE
Cannot change line types when plotting 1D functions

### ISSUE THREAD

**Issue by alexanderzimmerman** [firedrakeproject/firedrake#1387]
We (at least @wence, also Diego Volpatto but I don't see him here) briefly discussed this on Slack about a month ago and I had meant to make an issue out of it. Here's a MWE: 

```python
import firedrake as fe
import matplotlib.pyplot as plt


fig, axes = plt.subplots()

mesh = fe.UnitIntervalMesh(2)

P1 = fe.FiniteElement("P", mesh.ufl_cell(), 1)

V = fe.FunctionSpace(mesh, P1)

x = fe.SpatialCoordinate(mesh)[0]

for f, plot, tag in zip(
        ((0, 1), fe.interpolate(x, V)),
        (plt.plot, fe.plot),
        ("pyplot", "firedrake")):

    fig, axes = plt.subplots()
    
    plot(f, axes = axes, color = "r")
    
    plt.savefig("{0}.png".format(tag))

    print("Counted {0} lines from {1}".format(len(axes.get_lines()), tag))
```

which outputs firedrake.png

![image](https://user-images.githubusercontent.com/9367569/54068959-7ccbe700-4252-11e9-8e07-5a5d11d4e7da.png)

and pyplot.png

![image](https://user-images.githubusercontent.com/9367569/54068962-85242200-4252-11e9-810c-2ed09ff16452.png)

So the specific issue here is that the line in firedrake.png is the wrong color, and I haven't found any straightforward way of changing this. I think that the issue extends to anything else you'd like to do with the line type.

Furthermore, that script prints to stdout

> Counted 1 lines from pyplot
> Counted 0 lines from firedrake

demonstrating that, counterintuitively, firedrake.plot does not put any line objects onto the axes. Ideally I could just grab such lines and manipulate them directly.

**Comment by danshapero** [comment]
@alexanderzimmerman Can you try out the new plotting interface? In order to support plotting both discontinuous and high-degree polynomials in 1D, the `plot` function instead draws a matplotlib [`PathPatch`](https://matplotlib.org/api/_as_gen/matplotlib.patches.PathPatch.html) object. Any extra keyword arguments you pass on will go to the PathPatch constructor, so to change the color you'll probably want the `facecolor` argument.

**Comment by alexanderzimmerman** [comment]
I just ran the exact script from above with Firedrake that I built on 2020-03-13 (after much of your changes to the plot interface but I do see some changes came after), and both lines are now red as expected.

Also it's been working great for my various 2D plots. I had to change a bit of my user code, but it's overall a better design now and it was easy to use the new interface.

Thanks for your work!
